### PR TITLE
Initial setup of broadstreet ad package

### PR DESCRIPTION
**Pull request by B77Mills**
Add Broadstreet ad package with the ability to enable/disable on a per environment bases.  

Sample Config: 
```
module.exports = ({
  networkId: 7652,
  enable,
  zones: {
    billboard: {
      zoneIdSizeMapping: [
        ['', 99650],
      ],
    },
    leaderboard: {
      zoneIdSizeMapping: [
        ['(min-width: 980px)', 99648],
        ['(max-width: 980px)', 147846],
      ],
    },
    rotation: {
      zoneIdSizeMapping: [
        ['(min-width: 980px)', 99649],
        ['(max-width: 980px)', 147422],
      ],
    },
   // Only setup to handle placing in-body leaderboards(desktop/mobile) in different places.  
    // $ const desktopCounts = [1000, 2750, 4500, 6250, 8000, 9750, 11500, 13250, 15000, 16750, 18500];
    // $ const mobileCounts = [900, 1650, 2950, 4250, 5550, 6850, 8150, 9450, 10750, 12050, 13350, 14650, 15950, 17250];
    // These are called in [packages/global/components/layouts/content/components/content-body-with-injection.marko](https://github.com/parameter1/ab-media-websites/pull/218/files#diff-26787cfe1e9b39f4287e92cc925768006f0fcd18c179308b8b7dbbf0a736b8ec)
    contentDesktopRotation: {
      zoneIdSizeMapping: [
        ['(min-width: 980px)', 99649],
      ],
    },
    contentMobileLeaderboard: {
      zoneIdSizeMapping: [
        ['(max-width: 980px)', 147846],
      ],
    },
    contentMobileRotation: {
      zoneIdSizeMapping: [
        ['(max-width: 980px)', 147422],
      ],
    },
  },
});
```

At the moment I am not sure we need to have three or two media queries we will need to reach out to the group and confirm next steps.  

**At the moment the configs are setup to render their own ads at desktop resolution and a siblings site ads on non- desktop resolutions.  We will have to work with the group to setup non desktop ad zones for leaderboard and rotation zones at minimum.  Ww'll also have to work with them to update how they are setting up their ad targeting within Broadstreet.**

@TODO: figure out keyword targeting support(https://information.broadstreetads.com/how-keywords-work/).

### CODE REVIEW

File: packages/broadstreet/components/init.marko [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+** "useAltZone": function () { return false; },
+** "useZoneAliases": false,
+** "zoneOptions": {}       // options for an individual zone (overrides global options)
+**
+*/
+$ const src = input.src || "https://cdn.broadstreetads.com/init-2.min.js";
+$ const { on, config } = input;
+
+$ const createContainerScript = `
+  window.broadstreet = window.broadstreet || { run: [] };
```
Comment (zarathustra323) [inline packages/broadstreet/components/init.marko:29]: Change this to a single line:
```
$ const createContainerScript = `window.broadstreet = window.broadstreet || { run: [] }; window.broadstreet.run.push(function() { broadstreet.watch(${ JSON.stringify(config) }); });`;
```

File: packages/broadstreet/components/zone.marko [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+$ const sizeMapping = zoneIdSizeMapping || zone.zoneIdSizeMapping;
+$ const containerId = randomElementId({ prefix: 'div-broadstreet-ad' });
+
+<div
+  class=[blockName, ...BEM.applyModifiers(blockName, modifiers), input.class]
+  data-bam-zone-alias=zoneAlias
+>
+  <div id=containerId class=`${blockName} ${blockName}__wrapper` />
+  <script>
+    defineBroadstreetZone('${containerId}', ${JSON.stringify(sizeMapping)});
```
Comment (zarathustra323) [inline packages/broadstreet/components/zone.marko:21]: Should the first argument be as follows:
```js
defineBroadstreetZone(`'${containerId}'`, ${JSON.stringify(sizeMapping)});
```

File: packages/global/browser/billboard-cookie.vue [hunk 3]
```diff
@@ -6,6 +6,10 @@ import cookies from 'js-cookie';
 
 export default {
   props: {
+    adManagerType: {
```
Comment (zarathustra323) [inline packages/global/browser/billboard-cookie.vue:9]: Create two cookie files, one for GAM and one for BAM, and then conditionally include the file based on the configured ad server

**Review by zarathustra323 — CHANGES_REQUESTED** [review]

File: packages/broadstreet/components/zone.marko (hunk 2 above)
Comment (B77Mills) [reply packages/broadstreet/components/zone.marko:21]: @zarathustra323  I fought with this for a while and then found an example of rendering a function call within a script tag in marko here https://github.com/parameter1/base-cms/blob/master/packages/marko-web-gam/components/display-ad.marko on line 30

**Pull request closed without merging**